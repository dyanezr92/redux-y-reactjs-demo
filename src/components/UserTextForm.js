import React from 'react'
import PropTypes from 'prop-types'

class UserTextForm extends React.Component {
  constructor(props) {
    super(props);
    this.author = React.createRef();
    this.content = React.createRef();
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.onUserTextSubmit({
      text: this.content.current.value,
      author: this.author.current.value
    });
    this.content.current.value = "";
    this.author.current.value = "";
  }

  render(){
    return <form onSubmit={this.handleSubmit}>

      <div className="form-group">
        <label>Your { this.props.elementName }</label>
        <textarea ref={ this.content } className="form-control"/>
      </div>

      <div className="form-group">
        <label>Your name</label>
        <input type="text" ref={ this.author } className="form-control" />
      </div>

      <div>
        <button type="submit" className="btn btn-default">Send the { this.props.elementName }!</button>
      </div>
    </form>
  };
};

UserTextForm.propTypes = {
  elementName: PropTypes.string.isRequired,
  onUserTextSubmit: PropTypes.func.isRequired
}

export default UserTextForm
