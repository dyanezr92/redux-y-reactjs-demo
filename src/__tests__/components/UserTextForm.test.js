import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import UserTextForm from '../../components/UserTextForm';

describe('<UserTextForm onUserTextSubmit={ console.log } elementName="comment" />', () => {
  describe('render()', () => {
    test('renders the user form', () => {
      const wrapper = shallow(
        <UserTextForm onUserTextSubmit={ console.log } elementName="comment" />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('onSubmit()', () => {
    test('successfully calls the onSubmit handler', () => {
      const testValues = {
        author: '',
        text: '',
        handleSubmit: jest.fn(),
    };

    it('Submit works', () => {
        const component = shallow(
            <UserTextForm onUserTextSubmit={ console.log } elementName="comment" />
        );
        component.find('button').simulate('click');
        expect(testValues.handleSubmit).toHaveBeenCalledTimes(1);
        expect(testValues.handleSubmit).toBeCalledWith({author: testValues.author, text: testValues.text});
    });
    });
  });
});
