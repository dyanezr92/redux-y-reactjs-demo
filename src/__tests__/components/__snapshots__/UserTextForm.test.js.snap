// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserTextForm onUserTextSubmit={ console.log } elementName="comment" /> render() renders the user form 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label>
      Your 
      comment
    </label>
    <textarea
      className="form-control"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Your name
    </label>
    <input
      className="form-control"
      type="text"
    />
  </div>
  <div>
    <button
      className="btn btn-default"
      type="submit"
    >
      Send the 
      comment
      !
    </button>
  </div>
</form>
`;
